daemon/api_snap_conf.go

```diff
@@ -91,6 +97,58 @@ func getSnapConf(c *Command, r *http.Request, user *auth.UserState) Response {
 	return SyncResponse(currentConfValues)
 }
 
+// pruneExperimentalFlags returns an exact copy of val after hiding experimental
+// features that are no longer supported from core.experimental configuration
+// generic queries (i.e. key is "" or "experimental").
+// Exact queries (e.g. core.experimental.old-flag) are not pruned to avoid breaking
+// snaps that gate some behaviour behind a flag check.
+//
+// This helper should only be called for core configurations. Any errors when parsing
+// core config are ignored and val is returned without modification.
+func pruneExperimentalFlags(key string, val interface{}) interface{} {
```

Comment:
Simplification for consideration:
![image](https://github.com/user-attachments/assets/79ee089a-824b-4d96-bc4c-8188ae3b227a)